src/main/java/br/com/zup/concessionariacitroen/services/VendaService.java

```diff
@@ -0,0 +1,49 @@
+package br.com.zup.concessionariacitroen.services;
+
+import br.com.zup.concessionariacitroen.models.Pedido;
+import br.com.zup.concessionariacitroen.models.Venda;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class VendaService {
+
+    private List<Venda> listaVendas = new ArrayList<>();
+    @Autowired
+    PedidoService pedidoService;
+
+    public void efetuarVenda(Venda venda) {
+        listaVendas.add(venda);
+    }
+
+    public List<Venda> listarVendas() {
+        return listaVendas;
+    }
+
+    public List <Venda> pesquisarVendaPeloCPFdoCliente(String cpf) {
+        List<Venda> vendasDoClienteInformado = new ArrayList<>();
+
+        verificarListaVendasPorClienteEstaVazia();
+
+        for (Venda objetoVenda : listaVendas){
+
+            if (objetoVenda.getNumeroPedido().getCliente()
+                           .getCpf().equals(cpf)) {
+
+                vendasDoClienteInformado.add(objetoVenda);
+
+            }
+        }
+
+        return vendasDoClienteInformado;
+    }
+
+    private void verificarListaVendasPorClienteEstaVazia(){
+        if (listaVendas.isEmpty() || listaVendas == null){
+            throw new RuntimeException("A lista de vendas está vazia");
```

Comment:
Deve ser adicionado uma exception especial. Acredito que a mensagem de lista não é legal.
```suggestion
            throw new NaoHaVendasCadastradasException("Não há nenhuma venda cadastrada");
```